services/src/datasets/external/aruna/mod.rs

```diff
@@ -212,106 +217,299 @@ impl ArunaDataProvider {
-    async fn get_collection_info(&self, collection_id: String) -> Result<ArunaDatasetIds> {
-        let mut object_group_stub = self.object_group_stub.clone();
+    fn has_filter_label(&self, key_values: &Vec<KeyValue>) -> bool {
+        let label = &self.label_filter;
 
-        let mut aruna_object_group_overview = object_group_stub
-            .get_object_groups(GetObjectGroupsRequest {
-                collection_id: collection_id.clone(),
-                page_request: None,
-                label_id_filter: self.label_filter.clone(),
-            })
+        if let Some(filter) = label {
+            for key_value in key_values {
+                if key_value.variant() == KeyValueVariant::Label
+                    && key_value.key == *filter
+                    && key_value.value == *filter
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+        true
+    }
+
+    async fn get_verified_datasets(&self, dataset_ids: Vec<String>) -> Result<Vec<Dataset>> {
+        let mut dataset_stub = self.dataset_stub.clone();
+
+        let requested_datasets = dataset_stub
+            .get_datasets(GetDatasetsRequest { dataset_ids })
             .await?
             .into_inner()
-            .object_groups
-            .ok_or(ArunaProviderError::MissingObjectGroup)?
-            .object_group_overviews;
+            .datasets;
 
-        if aruna_object_group_overview.is_empty() {
-            return Err(ArunaProviderError::MissingObjectGroup);
-        }
+        let mut verified_datasets = vec![];
 
-        if aruna_object_group_overview.len() != 1 {
-            return Err(ArunaProviderError::UnexpectedObjectHierarchy);
+        for dataset in requested_datasets {
+            if self
+                .verify_resource(&dataset.id, &dataset.key_values, dataset.status())
+                .is_ok()
+            {
+                let outgoing_objects = Self::get_outgoing_internal_relation_ids(
+                    &dataset.relations,
+                    ResourceVariant::Object,
+                )
+                .ok();
+                if let Some(object_ids) = outgoing_objects {
+                    if self.get_verified_dataset_objects(object_ids).await.is_ok() {
+                        verified_datasets.push(dataset);
+                    }
+                }
+            }
         }
-        let object_group_id = aruna_object_group_overview
-            .pop()
-            .expect("Object groups should have size one")
-            .id;
-
-        let aruna_objects = object_group_stub
-            .get_object_group_objects(GetObjectGroupObjectsRequest {
-                collection_id: collection_id.clone(),
-                group_id: object_group_id.clone(),
-                page_request: None,
-                meta_only: false,
-            })
+        // //TODO: Workaround until deleted access permissions are fixed.
```

Comment:
what needs to be fixed there?

Reply:
https://github.com/geo-engine/geoengine/pull/946/commits/e114b50df3d522a4f8ee5e84d08ab4389bce4b4a

Forgot to delete the comment.

There was a bug in Aruna when querying deleted datasets that needed the code in the comment as a workaround, but I think it is fixed by now.